### PR TITLE
Svc conn inactive check before control evaluation

### PR DESCRIPTION
**Pull request by v-soukum**
## Description
</br>
Svc conn inactive check before control evaluation.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1 [hunk 1]
```diff
@@ -460,10 +461,24 @@ class ADOSVTBase: SVTBase {
 			#added check azuretable check here, if ((azuretable is used for storing bug info and scan mode is CA) OR azuretable bug info is disabed) then only allow bug logging
 			$scanSource = [AzSKSettings]::GetInstance().GetScanSource();
 			$isAzureTableEnabled = [Helpers]::CheckMember($this.ControlSettings.BugLogging, "UseAzureStorageAccount");
+			
+			#if bug logging is enabled for inactive resources, then only bug will be logged for inactive resources.
+			if([Helpers]::CheckMember($this.ControlSettings.BugLogging, "BugLogForInactiveResources"))
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1:466]: check member will only check whether property exists or not. we should also check whether it is enabled or not. Also t should be renamed to "LogBugsForInactiveResources".

File: src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1 [hunk 2]
```diff
@@ -2,6 +2,7 @@ class ADOSVTBase: SVTBase {
 
 	hidden [ControlStateExtension] $ControlStateExt;
 	hidden [AzSKSettings] $AzSKSettings;
+	hidden [bool] $isResourceActive = $true;
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1:5]: This should be reset for each resource individually.

File: src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1 [hunk 3]
```diff
@@ -460,10 +461,24 @@ class ADOSVTBase: SVTBase {
...
 			$scanSource = [AzSKSettings]::GetInstance().GetScanSource();
 			$isAzureTableEnabled = [Helpers]::CheckMember($this.ControlSettings.BugLogging, "UseAzureStorageAccount");
+			
+			#if bug logging is enabled for inactive resources, then only bug will be logged for inactive resources.
+			if([Helpers]::CheckMember($this.ControlSettings.BugLogging, "BugLogForInactiveResources"))
+			{
+				$bugLogForInactiveResources = $true;
+			}
+			else {
+				$bugLogForInactiveResources = $this.isResourceActive;
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1:471]: shouldn't this be dependent on individual rsrc status?

File: src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1 (hunk 2 above)
Comment (abhaydaga) [reply src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1:5]: add comment

File: src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1 [hunk 4]
```diff
@@ -460,10 +461,24 @@ class ADOSVTBase: SVTBase {
...
 			{
-				if (($ControlResults.ControlResults.VerificationResult -contains "Failed") -or ($ControlResults.ControlResults.VerificationResult -contains "Verify")) {
-					$this.BugLoggingPostEvaluation($ControlResults, $BugLogParameterValue)
+				if ($bugLogForInactiveResources) {
+					if (($ControlResults.ControlResults.VerificationResult -contains "Failed") -or ($ControlResults.ControlResults.VerificationResult -contains "Verify")) {
+						$this.BugLoggingPostEvaluation($ControlResults, $BugLogParameterValue)
+					}
+				}
+				else {
+					$this.PublishCustomMessage("Bug logging is disabled for inactive resources.", [MessageType]::Info);
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1:481]: The current resource is inactive. Bug logging is disabled for inactive resources.

Comment (abhaydaga) [reply src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1:481]: Warning

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1 [hunk 5]
```diff
@@ -575,4 +573,57 @@ class ServiceConnection: ADOSVTBase
...
+                        }
+                        else
+                        {
+                            $this.SvcConnActivityDetail.isSvcConnActive = $true;
+                            $this.SvcConnActivityDetail.message =  "Service connection has been used in the last $inactiveLimit days.";
+                        }
+                    }
+                    else {
+                        $this.SvcConnActivityDetail.isSvcConnActive = $false;
+                        $this.SvcConnActivityDetail.message = "History period in days (ServiceConnectionHistoryPeriodInDays) to check last running day of service connection is not defined in your organization policy. Please update your ControlSettings.json as per the latest AzSK.ADO PowerShell module.";
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1:608]: remove

Comment (abhaydaga) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1:608]: Could not fetch the inactive limit for service connection.

File: src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1 (hunk 1 above)
Comment (v-soukum) [reply src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1]: changes done

File: src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1 (hunk 2 above)
Comment (v-soukum) [reply src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1:6]: Value for this variable will be overridden for each and every resource

File: src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1 (hunk 4 above)
Comment (v-soukum) [reply src/AzSK.ADO/Framework/Abstracts/ADOSVTBase.ps1]: changes done

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1 (hunk 5 above)
Comment (v-soukum) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1]: message changed

**Review by abhaydaga — DISMISSED** [review]

**Review by abhaydaga — APPROVED** [review]

**Pull request merged**